src/main/java/org/opensearch/security/auth/internal/InternalAuthenticationBackend.java

```diff
@@ -157,7 +159,7 @@ public User authenticate(final AuthCredentials credentials) {
 
     @Override
     public String getType() {
```

Comment:
What do you think about updating the AuthorizationBackend interface?  Given a user they came from a kind of backend, such as ldap, but there is also the possibility of multiple backends with the same type.

Reply:
That's what I was thinking about when deciding to use the key in the `authc` section of the `config.yml` as the name of the auth domain that gets set on the user object.

So if an authc section is configured like this:

```
authc:
    basic_internal_auth_domain:
        http_enabled: true
        transport_enabled: true
        order: 0
        http_authenticator:
          type: basic
          challenge: false
        authentication_backend:
          type: internal
      openid_auth_domain:
        http_enabled: true
        transport_enabled: true
        order: 1
        http_authenticator:
          type: openid
          challenge: false
          config:
            subject_key: preferred_username
            roles_key: groups
            openid_connect_url: http://host.docker.internal:8080/auth/realms/master/.well-known/openid-configuration
        authentication_backend:
          type: noop
```

Then a user is either from the `basic_internal_auth_domain` or from `openid_auth_domain` and those would be the potential values that `User.getAuthDomain()` could return. `admin_certificate` is a special type of auth domain because client cert auth domain doesn't need to be configured to use admin_cert, only `plugins.security.authcz.admin_dn` in `opensearch.yml.

Reply:
> I don't think its clear to operators that changing the key could break functionality

I've been thinking about this problem too and I'm not sure how best to solve for it. Let's say we start keeping track of which users logged in, when they logged in and what auth backend authenticated the user. If a user changed the key to the auth backend in the config.yml then it would not be able to connect the users back to the previous name used if the name of the auth domain was changed. Is there a better way to connect the user to an auth domain in the list of authentication backends?